### PR TITLE
Commented and removed unused code for scripts and source files

### PR DESCRIPTION
**Pull request by NxFerrara**
All scripts and source files have been commented. 

Programs modified by removing unused code: read_ultralight.c, button_listener.c, and screen.py 

Please review the commented code to become more familiar. Identify areas of needed clarification, so I can make it clearer. Merge the branch if the comments are sensible.

### CODE REVIEW

File: scripts/monitor.sh [hunk 1]
```diff
@@ -1,96 +1,99 @@
 #!/bin/bash
 
-# Specify the base directory
+# Define the base directory for the NFC tracking application.
 BASE_DIR="/home/potato/NFC_Tracking"
 
-# Specify the name and path of your C programs
+# Define the names and paths of the C programs to be monitored.
 PROGRAM_NAME_1="read_ultralight"
```
Comment (jbesoto) [inline scripts/monitor.sh]: Consider using arrays to manage the program names and paths. This approach will make the script more concise, reduce potential errors from copy-pasting, and make it easier to add or remove programs in the future. 

```diff
- PROGRAM_NAME_1="read_ultralight"
- PROGRAM_NAME_2="button_listener"
- PROGRAM_NAME_3="screen.py"
- PROGRAM_PATH_1="$BASE_DIR/$PROGRAM_NAME_1"
- PROGRAM_PATH_2="$BASE_DIR/$PROGRAM_NAME_2"
- PROGRAM_PATH_3="$BASE_DIR/$PROGRAM_NAME_3"
```
Suggested change:
```diff
+ PROGRAM_NAMES=("read_ultralight" "button_listener" "screen.py")
+ for idx in ${!PROGRAM_NAMES[@]}; do
+    PROGRAM_PATHS[$idx]=("${BASE_DIR}/${PROGRAM_NAMES[$idx]}")
```

File: src/button_listener.c [hunk 2]
```diff
@@ -3,41 +3,57 @@
...
+struct gpiod_line *button_line; // GPIO line struct
 
-volatile sig_atomic_t interrupted = 0;
+volatile sig_atomic_t interrupted = 0; // Flag checking for whether the program was interrupted.
 
 void cleanup_gpio(void);
 
-// This function will be called when SIGINT is sent to the program (Ctrl+C)
 void handle_sigint(int sig) {
+    /**
```
Comment (jbesoto) [inline src/button_listener.c]: Use Doxygen-style to comment functions. Here is an example from GPT:
```c
/**
 * @brief Computes the area of a rectangle.
 *
 * Given the width and height of a rectangle, this function calculates and
 * returns its area.
 *
 * @param width The width of the rectangle. Must be non-negative.
 * @param height The height of the rectangle. Must be non-negative.
 * 
 * @return The area of the rectangle. Returns -1 if any parameter is negative.
 *
 * @note This function does not check for arithmetic overflow.
 */
double computeRectangleArea(double width, double height) {
    if (width < 0 || height < 0) {
        return -1.0;  // Error value
    }
    return width * height;
}
```
The `@brief` summarizes the function while `@note` are additional notes or warnings about the function (optional).

File: src/read_ultralight.c [hunk 3]
```diff
@@ -15,18 +12,17 @@
...
-
-struct gpiod_chip *chip;
-struct gpiod_line *red_line, *yellow_line, *green_line;
-
 const char *fifo_path = "/tmp/screenPipe";
 
-// The function that logs messages with timestamp
 void print_log(const char *format, ...) {
+    /**
+     * Function: print_log
```
Comment (jbesoto) [inline src/read_ultralight.c]: Use Doxygen-style to document the functions.

File: src/screen.py [hunk 4]
```diff
@@ -21,13 +21,30 @@
 
 
 def handle_sigterm(signum, frame):
+    """
```
Comment (jbesoto) [inline src/screen.py]: Use Google Style for commenting functions. More information [here](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods).

File: src/dynamodb.py [hunk 5]
```diff
@@ -6,17 +6,29 @@
 import os
 import argparse
 
-
 def delete_all_items(table):
+    """
```
Comment (jbesoto) [inline src/dynamodb.py]: Use Google-style to comment functions. More information [here](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods)

**Review by jbesoto — CHANGES_REQUESTED** [review]
Good, meaningful comments. Fix minor details.

File: scripts/monitor.sh (hunk 1 above)
Comment (kc34251) [reply scripts/monitor.sh]: Looks good

**Comment by kc34251** [comment]
Looks good

**Comment by NxFerrara** [comment]
The most recent commit to this branch may cause some merge conflicts with the refactor/repo-cleanup branch since its most recent commit refactored by reorganizing some src and scripts files while renaming some files as well. Just wanted to mention this for when merge conflict occurs. Not sure if there is a better way of dealing with this like reverting the changes of the last commit on this branch or just dealing with it during merging.

File: scripts/monitor.sh [hunk 6]
```diff
@@ -51,37 +52,37 @@ while true; do
...
         # If not running, try to restart the program and log its output.
-        $PROGRAM_PATH_1 >> /var/log/programs.log 2>&1 &
+        $PROGRAM_PATHS[0] >> /var/log/programs.log 2>&1 &
         # Store the PID of the restarted program.
         echo $! > /var/run/read_ultralight.pid
     fi
 
     # Check if the second program is running.
-    if pgrep -f $PROGRAM_NAME_2 > /dev/null
+    if pgrep -f $PROGRAM_NAMES[2] > /dev/null
```
Comment (NxFerrara) [inline scripts/monitor.sh]: Owner, so I can't make a suggestion directly, but here is my suggestion:
`if pgrep -f $PROGRAM_NAMES[1] > /dev/null`

File: scripts/monitor.sh [hunk 7]
```diff
@@ -4,12 +4,13 @@
...
-PROGRAM_NAME_1="read_ultralight"
-PROGRAM_NAME_2="button_listener"
-PROGRAM_NAME_3="screen.py"
-PROGRAM_PATH_1="$BASE_DIR/$PROGRAM_NAME_1"
-PROGRAM_PATH_2="$BASE_DIR/$PROGRAM_NAME_2"
-PROGRAM_PATH_3="$BASE_DIR/$PROGRAM_NAME_3"
+PROGRAM_NAMES=(
+    "nfc_tap_listener.c"
+    "operation_tap_listener.c"
+    "screen.py"
```
Comment (NxFerrara) [inline scripts/monitor.sh]: Owner, so I can't make a suggestion directly, but here is my suggestion:
`"display_and_data_handler.py"`

File: scripts/monitor.sh (hunk 6 above)
Comment (jbesoto) [reply scripts/monitor.sh]: Thanks for pointing it out!

**Comment by jbesoto** [comment]
Pending update comment format in Python files. I will take care of it.

**Review by jbesoto — APPROVED** [review]
Approving comment format. Changes previously discussed and reviewed in-person with @NxFerrara.

**Review by jbesoto — APPROVED** [review]
- [x] Environment variables used
- [x] Commenting following team guidelines
- [x] Remove unnecessary/unused code (create new issue to fix)

**Review by jbesoto — APPROVED** [review]
Ignore this. Changes already approved

**Pull request merged**